core_impl/src/test/java/com/google/instrumentation/tags/TagSetTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.tags;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.instrumentation.internal.StringUtil;
+import com.google.instrumentation.internal.logging.TestLogger;
+import com.google.instrumentation.internal.logging.TestLogger.LogRecord;
+import java.util.Arrays;
+import java.util.logging.Level;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests for {@link TagSet}. */
+// TODO(sebright): Add more tests once the API is finalized.
+@RunWith(JUnit4.class)
+public class TagSetTest {
+
+  private static final TagKey<String> KS1 = TagKey.createString("k1");
+  private static final TagKey<String> KS2 = TagKey.createString("k2");
+  private static final TagKey<Long> KI = TagKey.createInt("k2");
```

Comment:
Should add a test doing something similar to (same key name, different types):

newBuilder().set(KS2, "v1").set(KI, 5).build()